modules/api/test/LorisApiCandidatesTest.php

```diff
@@ -0,0 +1,307 @@
+        $this->assertNotEmpty($body);
+
+        $candidatesCandidArray = json_decode(
+            (string) utf8_encode(
+                $response->getBody()->getContents()
+            ),
+            true
+        );
+
+        $this->assertArrayHasKey(
+            'Meta',
+            $candidatesCandidArray
+        );
+        $this->assertArrayHasKey(
+            'Visits',
+            $candidatesCandidArray
+        );
+        $this->assertArrayHasKey(
+            '0',
+            $candidatesCandidArray['Visits']
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray),
+            'array'
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray['Meta']),
+            'array'
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray['Meta']['CandID']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray['Meta']['Project']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray['Meta']['PSCID']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray['Meta']['Site']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray['Meta']['EDC']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray['Meta']['DoB']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray['Meta']['Sex']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray['Visits']),
+            'array'
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray['Visits']['0']),
+            'string'
+        );
+
+    }
+
+    /**
+     * Tests the HTTP POST request for the endpoint /candidates/{candid}
+     *
+     * @return void
+     */
+    public function testPostCandidatesCandid(): void
+    {
+        // First, create a valid new candidate
+        $json_new     = [
+            'Candidate' =>
+                [
+                    'Project' => "Rye",
+                    'Site'    => "Data Coordinating Center",
+                    'EDC'     => "2020-01-03",
+                    'DoB'     => "2020-01-03",
+                    'Sex'     => "Male"
+                ]
+        ];
+        $response_new = $this->client->request(
+            'POST',
+            "candidates",
+            [
+                'headers' => $this->headers,
+                'json'    => $json_new
+            ]
+        );
+        // Verify the status code
+        $this->assertEquals(201, $response_new->getStatusCode());
+        // Verify the endpoint has a body
+        $body = $response_new->getBody();
+        $this->assertNotEmpty($body);
+
+        // Second, create a candidate that already exist
```

Comment:
What do you mean by already exists?

Reply:
I made a mistake, I meant to try creating a candidate that already exist, so actually I would not have to create `$json_exist` and just reuse `$json_new`. Now that I think about it, I'm not sure it is pertinent to test adding the same candidate twice?